### PR TITLE
Markdown - Fixed bug with local anchor links

### PR DESCRIPTION
**Pull request by Ixiter**
Since we use `<base>` in our html, all local `#anchor` links will lead to `System::baseUrl()`#anchor.

Therefore we cannot use `#anchors` in markdown.

### Example: 
The following is part of `/addon/myaddon/README.md` on my `https://example.com` so we can see it at `https://example.com/admin/addons/myaddon`

```markdown
## Contents
[Chapter 1](#chapter1)

### <a name="chapter1"></a>Chapter 1
```
#### Current behaviour
The link leads to `http://example.com/#chapter1`.

#### Expected behaviour
The link should lead to `https://example.com/admin/addons/myaddon#chapter1`

#### Solution
The REQUEST_URI was missing. I added it to all links using local `#anchors`

### CODE REVIEW

**Comment by Ixiter** [comment]
LOL .. I have no idea how my own merge is in this PR now. However .. its a no changes merge.

**Comment by MrPetovan** [comment]
If you merge your own branches in your own `develop` branch, then it will be different from the actual merge in the `friendica/develop` branch. Therefore, when you submit a new pull request, this merge on your own repo will be included in the commit list, even though no files are different. I suggest you don't merge your own branches into your `develop` branches, and instead rebase your develop branch onto `friendica/develop` as your PRs are merged. This ensures your `develop` branch is exactly the same as `friendica/develop`.

File: src/Content/Text/Markdown.php [hunk 1]
```diff
@@ -34,6 +34,7 @@ public static function convert($text, $hardwrap = true) {
 		$MarkdownParser->hard_wrap = $hardwrap;
 		$MarkdownParser->code_class_prefix = 'language-';
 		$html = $MarkdownParser->transform($text);
+		$html = str_replace('<a href="#', '<a href="' . ltrim($_SERVER['REQUEST_URI'], '/') . '#', $html);
```
Comment (MrPetovan) [inline src/Content/Text/Markdown.php]: Are you sure there aren't any `<a>` tag with attributes before the `href`?

Comment (Ixiter) [reply src/Content/Text/Markdown.php]: No, I am not sure. You are right, I will change it to a preg_replace.

Comment (MrPetovan) [reply src/Content/Text/Markdown.php]: Or maybe just `str_replace('href="#', 'href="' . ltrim($_SERVER['REQUEST_URI'], '/') . '#', $html)`

**Comment by Ixiter** [comment]
> If you merge your own branches in your own `develop` branch, then it will be different from the actual merge in the `friendica/develop` branch. Therefore, when you submit a new pull request, this merge on your own repo will be included in the commit list, even though no files are different. I suggest you don't merge your own branches into your `develop` branches, and instead rebase your develop branch onto `friendica/develop` as your PRs are merged. This ensures your `develop` branch is exactly the same as `friendica/develop`.

I did the following ..

On github I did a pullrequest to sync my github develop with the friendica develop
Then I pulled my github develop to my local develop. After I though its a good idea to pull my local develop to my local develop-sub-branches. I think for the last step I mixed up the parameters somehow.

still not familiar with this git commands

File: src/Content/Text/Markdown.php (hunk 1 above)
Comment (Ixiter) [reply src/Content/Text/Markdown.php]: hmm .. can we be sure, that this is always part of a html <a element ?

BTW .. I just found https://regex101.com/ which also includes a regex debugger. The debugger shows very good how regex works. nice.

**Pull request merged**